src/expression.js

```diff
@@ -1707,6 +1807,100 @@ function caml_binaryen_ref_eq(wasm_mod, left, right) {
   return wasm_mod.ref.func(left, right);
 }
 
+// Struct operations
+
+//Provides: caml_binaryen_struct_new
+//Requires: caml_list_to_js_array
+function caml_binaryen_struct_new(wasm_mod, operands, type) {
+  return wasm_mod.struct.new(
+    operands ? caml_list_to_js_array(operands[1]) : null,
+    type
+  );
+}
+
+//Provides: caml_binaryen_struct_get
+//Requires: caml_js_from_bool
+function caml_binaryen_struct_get(wasm_mod, index, ref, type, signed) {
+  if (caml_js_from_bool(signed)) {
+    return wasm_mod.struct.get_s(index, ref, type);
+  } else {
+    return wasm_mod.struct.get_u(index, ref, type);
+  }
+}
+
+//Provides: caml_binaryen_struct_set
+function caml_binaryen_struct_set(wasm_mod, index, ref, value) {
+  return wasm_mod.struct.set(index, ref, value);
+}
+
+// Array operations
+
+//Provides: caml_binaryen_array_new
+function caml_binaryen_array_new(wasm_mod, type, size, init) {
+  return wasm_mod.array.new(type, size, init);
+}
+
+//Provides: caml_binaryen_array_new_data
+//Requires: caml_jsstring_of_string
+function caml_binaryen_array_new_data(wasm_mod, type, name, offset, size) {
+  return wasm_mod.array.new_data(
+    type,
+    caml_jsstring_of_string(name),
+    offset,
+    size
+  );
+}
+
+//Provides: caml_binaryen_array_new_fixed
+//Requires: caml_list_to_js_array
+function caml_binaryen_array_new_fixed(wasm_mod, type, values) {
+  return wasm_mod.array.new_fixed(type, caml_list_to_js_array(values));
+}
+
+//Provides: caml_binaryen_array_get
+//Requires: caml_js_from_bool
+function caml_binaryen_array_get(wasm_mod, ref, index, type, signed) {
+  if (caml_js_from_bool(signed)) {
+    return wasm_mod.array.get_s(ref, index, type);
```

Comment:
In the v124 vindings it seems that this is just `array.get` which accepts a signed bool. 

https://github.com/WebAssembly/binaryen/blob/64ba23996a10e229d46e41eb37736a55af87f79a/src/js/binaryen.js-post.js#L2529